UM/Trust.py

```diff
@@ -435,6 +448,20 @@ def signedFolderCheck(self, path: str) -> bool:
                         if os.path.islink(dir_full_path) and not self._follow_symlinks:
                             Logger.log("w", "Directory symbolic link '{0}' will not be followed.".format(dir_full_path))
 
+                # Check if the files moved to storage are still correct.
+                if storage_json:
+                    for entry in storage_json:
+                        try:
+                            # If this doesn't raise an exception, it's correct.
+                            CentralFileStorage.retrieve(entry[1], entry[3], Version(entry[2]))
+                        except:
```

Comment:
It would be better to catch the specific exceptions that you expect here. That way, if we change the code in the future, we'll get a correct error message when we've made a mistake.
```suggestion
                        except (EnvironmentError, IOError):
```